sessionctx/variable/sysvar.go

```diff
@@ -660,7 +661,21 @@ var defaultSysVars = []*SysVar{
 		return nil
 	}},
 	{Scope: ScopeNone, Name: Hostname, Value: DefHostname},
-	{Scope: ScopeSession, Name: Timestamp, Value: "", skipInit: true},
+	{Scope: ScopeSession, Name: Timestamp, Value: "0", skipInit: true, MinValue: 0, MaxValue: 2147483647, Type: TypeFloat, GetSession: func(s *SessionVars) (string, error) {
+		defaultTimestamp := GetSysVar(Timestamp).Value
+		if timestamp, ok := s.systems[Timestamp]; ok && timestamp != defaultTimestamp {
+			return timestamp, nil
+		}
+		timestamp := s.StmtCtx.GetOrStoreStmtCache(stmtctx.StmtNowTsCacheKey, time.Now()).(time.Time)
+		return types.ToString(float64(timestamp.UnixNano()) / float64(time.Second))
+	}, GetGlobal: func(s *SessionVars) (string, error) {
+		// The Timestamp sysvar will have GetGlobal func even though it does not have global scope.
+		// It's GetGlobal func will only be called when "set timestamp = default".
+		// Setting timestamp to DEFAULT causes its value to be the current date and time as of the time it is accessed.
+		// See https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_timestamp
+		defaultTimestamp := GetSysVar(Timestamp).Value
+		return defaultTimestamp, nil
+	}},
```

Comment:
I think we should eliminate the defaultTimestamp by using a constant (`DefTimestamp`) that you can define in tidb_vars.go.

```suggestion
	{Scope: ScopeSession, Name: Timestamp, Value: DefTimestamp, skipInit: true, MinValue: 0, MaxValue: 2147483647, Type: TypeFloat, GetSession: func(s *SessionVars) (string, error) {
		if timestamp, ok := s.systems[Timestamp]; ok && timestamp != DefTimestamp {
			return timestamp, nil
		}
		timestamp := s.StmtCtx.GetOrStoreStmtCache(stmtctx.StmtNowTsCacheKey, time.Now()).(time.Time)
		return types.ToString(float64(timestamp.UnixNano()) / float64(time.Second))
	}, GetGlobal: func(s *SessionVars) (string, error) {
		// The Timestamp sysvar will have GetGlobal func even though it does not have global scope.
		// It's GetGlobal func will only be called when "set timestamp = default".
		// Setting timestamp to DEFAULT causes its value to be the current date and time as of the time it is accessed.
		// See https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_timestamp
		return DefTimestamp, nil
	}},
```

Otherwise LGTM.